app/features/send/send-store.ts

```diff
@@ -217,26 +242,38 @@ export const createSendStore = ({
       status: 'idle',
       amount: null,
       accountId: initialAccount.id,
-      sendType: 'CASHU_TOKEN',
+      sendType: getDefaultSendType(initialAccount.type),
       destination: null,
       destinationDisplay: null,
       quote: null,
       cashuToken: null,
 
-      selectSourceAccount: (account) => set({ accountId: account.id }),
+      selectSourceAccount: (account) => {
+        const { destination, destinationDetails, sendType } = get();
+        const hasDestination = !!destination || !!destinationDetails;
+        set({
+          accountId: account.id,
+          sendType: hasDestination
+            ? sendType
```

Comment:
so if cashu account was first selected and then changed to spark and destination was selected, wouldn't this possibly leave invalid sendType?